matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/create/CreateRoomTask.kt

```diff
@@ -55,10 +57,26 @@ internal class DefaultCreateRoomTask @Inject constructor(
 ) : CreateRoomTask {
 
     override suspend fun execute(params: CreateRoomParams): String {
+        val otherUserId = if (params.isDirect()) {
+            params.getFirstInvitedUserId()
+                    ?: throw IllegalStateException("You can't create a direct room without an invitedUser")
+        } else null
+
         val createRoomBody = createRoomBodyBuilder.build(params)
 
-        val createRoomResponse = executeRequest<CreateRoomResponse>(eventBus) {
-            apiCall = roomAPI.createRoom(createRoomBody)
+        val createRoomResponse = try {
+            executeRequest<CreateRoomResponse>(eventBus) {
+                apiCall = roomAPI.createRoom(createRoomBody)
+            }
+        } catch (throwable: Throwable) {
+            if (throwable is Failure.ServerError
+                    && throwable.httpCode == 403
+                    && throwable.error.code == MatrixError.M_FORBIDDEN
+                    && throwable.error.message.startsWith("Federation denied with")) {
```

Comment:
I didn't like this but I guess there isn't any other way to handle...

Reply:
Yes, I suppose. This whole thing is a big workaround which may be improved server side in the future.